api/models/Group.js

```diff
@@ -533,6 +541,23 @@ module.exports = bookshelf.Model.extend(merge({
   },
 
   // ******* Class methods ******** //
+
+  // Background task to do additional work/tasks when new members are added to a group
+  async afterAddMembers({ groupId, newUserIds, reactivatedUserIds }) {
+    const zapierTriggers = await ZapierTrigger.query(q => q.where({ group_id: groupId, type: 'new_member' })).fetchAll()
+    if (zapierTriggers && zapierTriggers.length > 0) {
+      const members = await User.query(q => q.whereIn('id', newUserIds.concat(reactivatedUserIds))).fetchAll()
+      for (const trigger of zapierTriggers) {
+        const response = await fetch(trigger.get('target_url'), {
+          method: 'post',
+          body: JSON.stringify(members.map(m => ({ id: m.id, name: m.get('name'), reactivated: reactivatedUserIds.includes(m.id) }))),
+          headers: { 'Content-Type': 'application/json' }
+        })
+        // TODO: what to do with the response? check if succeeded or not?
```

Comment:
Still valid?

Reply:
yeah i will leave this here for now until we have real world experience with Zapier and know if we need to do anything here